assertk/src/commonMain/kotlin/assertk/assert.kt

```diff
@@ -190,3 +191,13 @@ inline fun assertFailure(f: () -> Unit): Assert<Throwable> {
     }
     fail("expected failure but lambda completed successfully")
 }
+
+/**
+ * Asserts that the given block will throw an exception with expected type rather than complete successfully.
+ */
+inline fun <reified T: Throwable> assertFailureWith(f: () -> Unit): Assert<T> {
+    val assertFailure = assertFailure(f)
+    assertFailure.isInstanceOf(T::class)
+    @Suppress("UNCHECKED_CAST")
+    return assertFailure as Assert<T>
```

Comment:
This returns a specialized `Assert` with the new type
```suggestion
    return assertFailure.isInstanceOf(T::class)
```